test_utils/__init__.py

```diff
@@ -116,4 +118,35 @@ def set_jwt_cookie(self, roles_and_contexts=[]):
         })
         jwt_token = generate_jwt_token(payload)
 
-        self.client.cookies[jwt_cookie_name()] = jwt_token
\ No newline at end of file
+        self.client.cookies[jwt_cookie_name()] = jwt_token
+
+
+class APITestWithMockedDiscoveryApiClient(APITest):
+    """
+    API test class with discovery api calls in subsidy_request tasks mocked out.
+
+    We call discovery on every SubsidyRequest object save().
+    """
+    def setUp(self):
+        super().setUp()
+        self.disco_patcher = mock.patch('enterprise_access.apps.subsidy_request.tasks.DiscoveryApiClient')
+        self.mock_discovery_client = self.disco_patcher.start()
+        self.mock_discovery_client().get_course_data.return_value = {
+            'title': 'How to Bake a Pie: A Slice of Heaven',
+        }
+        self.addCleanup(self.disco_patcher.stop)
+
+
+class TestCaseWithMockedDiscoveryApiClient(TestCase):
+    """
+    Test class with discovery api calls in subsidy_request tasks mocked out.
+
+    We call discovery on every SubsidyRequest object save().
+    """
+    def setUp(self):
+        super().setUp()
+        self.disco_patcher = mock.patch('enterprise_access.apps.subsidy_request.tasks.DiscoveryApiClient')
+        self.mock_discovery_client = self.disco_patcher.start()
+        self.mock_discovery_client().get_course_data.return_value = {
+            'title': 'How to Bake a Cake: So Delicious It Should Be Illegal',
```

Comment:
nit: Would it be helpful to have this example title as a constant for re-use in test assertions elsewhere?

Reply:
I can make a descriptive variable name for the best of both worlds ;)